### PR TITLE
Update documentation based on feedback

### PR DESCRIPTION
**Pull request by sushshring**

### CODE REVIEW

File: docs/mkdocs/docs/api/macros/json_diagnostic_positions.md [hunk 1]
```diff
@@ -10,9 +10,12 @@ When enabled, two new properties: `start_pos()` and `end_pos()` are added to `nl
 that JSON object/field in the original string the object was parsed from. Likewise, `end_pos()` returns the end position of that JSON object/field in the
 original string the object was parsed from.
 
-For objects and arrays, the start and end positions represent the positions of the opening and closing braces or brackets, respectively. For fields, the start
-and end positions represent either the opening and closing quotes for that field's value or the first and character after last in the field's numerical or
-predefined true/false/null values.
+For objects and arrays, the start position represents the position of the opening brace or bracket. The end position is the character after the closing
```
Comment (raphgianotti) [inline docs/mkdocs/docs/api/macros/json_diagnostic_positions.md]: I will leave it up to you, but I prefer a version of separating talking about what start/end means vs what the first and last character is for each type.

File: docs/mkdocs/docs/api/macros/json_diagnostic_positions.md [hunk 2]
```diff
@@ -6,17 +6,20 @@
...
 
-When enabled, two new properties: `start_pos()` and `end_pos()` are added to `nlohmann::json` objects and fields. `start_pos()` returns the start position of
+When enabled, two new properties: `start_pos()` and `end_pos()` are added to `nlohmann::basic_json` objects and fields. `start_pos()` returns the start position of
 that JSON object/field in the original string the object was parsed from. Likewise, `end_pos()` returns the end position of that JSON object/field in the
 original string the object was parsed from.
 
-For objects and arrays, the start and end positions represent the positions of the opening and closing braces or brackets, respectively. For fields, the start
-and end positions represent either the opening and closing quotes for that field's value or the first and character after last in the field's numerical or
-predefined true/false/null values.
+For objects and arrays, the start position represents the position of the opening brace or bracket. The end position is the character after the closing
```
Comment (raphgianotti) [inline docs/mkdocs/docs/api/macros/json_diagnostic_positions.md]: One nit with the current version, we use a period between the start/end explanations for objects/arrays, and a comma for the fields version, should be consistent

File: docs/mkdocs/docs/api/macros/json_diagnostic_positions.md [hunk 3]
```diff
@@ -6,17 +6,20 @@
...
+brace or bracket. For fields, the start position represents the opening quote or first character in field's numerical or predefined (true/false/null) value,
+the end position represents the character after the closing quote or the character after the last character in the field's numerical or predefined value.
 
-`start_pos()` and `end_pos()` are only set if the JSON object was parsed using `json::parse()`. For all other cases, `std::string::npos` will be returned.
+Given the above, `end_pos() - start_pos()` for an object or field provides the length of the string representation for that object or field, including the
```
Comment (raphgianotti) [inline docs/mkdocs/docs/api/macros/json_diagnostic_positions.md:15]: Just double checking if the back ticks should include the operation or just the operands. Dunno what the standard they use elsewhere is.

Comment (sushshring) [reply docs/mkdocs/docs/api/macros/json_diagnostic_positions.md:15]: They didn't seem to have an issue with the operator being in the backticks, I will leave it this way

**Review by raphgianotti — APPROVED** [review]

**Pull request merged**